sklearnex/covariance/tests/test_incremental_covariance.py

```diff
@@ -37,6 +38,14 @@
 @pytest.mark.parametrize("dtype", [np.float32, np.float64])
 @pytest.mark.parametrize("assume_centered", [True, False])
 def test_sklearnex_partial_fit_on_gold_data(dataframe, queue, dtype, assume_centered):
+    try:
+        from dpctl import device_type
+
+        is_gpu = queue is not None and queue.sycl_device.device_type == device_type.gpu
```

Comment:
doesn't there exist an is_gpu attribute on a sycl_device (https://github.com/intel/scikit-learn-intelex/blob/main/onedal/_device_offload.py#L50)? Should allow you to remove the try except and remove the codefactor issue with the ```except```